thrust/thrust/system/cuda/detail/transform_scan.h

```diff
@@ -69,6 +69,26 @@ OutputIt _CCCL_HOST_DEVICE transform_inclusive_scan(
   return cuda_cub::inclusive_scan_n(policy, transformed_iterator_t(first, transform_op), num_items, result, scan_op);
 }
 
+template <class Derived, class InputIt, class OutputIt, class TransformOp, class InitialValueType, class ScanOp>
+OutputIt _CCCL_HOST_DEVICE transform_inclusive_scan(
+  execution_policy<Derived>& policy,
+  InputIt first,
+  InputIt last,
+  OutputIt result,
+  TransformOp transform_op,
+  InitialValueType init,
+  ScanOp scan_op)
+{
+  using result_type = thrust::remove_cvref_t<InitialValueType>; // cuda::std::__accumulator_t?
```

Comment:
Lets just use `__accumulator_t`

Reply:
`__accumulator_t` is better than `InitialValueType`. But IMO, even better would be `invoke_result<TransformOp, value_t<InputIt>>` (please check exact spelling). I am even surprised we need to tell `transform_input_iterator_t` what its value type is, since it could just infer it from `TransformOp`.

Reply:
As discussed in https://github.com/NVIDIA/cccl/pull/1940#discussion_r1701299395, we should use accumulator type in new API

Reply:
We discussed using `__accumulator_t` for the intermediate results of the accumulation. The type `result_type` here is the type returned from the transformation before accumulation even starts.

Spelling out the accumulation `scan_op(scan_op(scan_op(init, transform_op(in[0])), transform_op(in[1])), transform_op(in[2])), ...)`, we are talking about the type of `transform_op(...)`.

Reply:
agreed (also Georgii outlines the same idea further down)